### PR TITLE
Add guide for `ActiveRecord::DelegatedType` [skip ci]

### PR DESCRIPTION
**Pull request by soartec-lab**
### Motivation / Background

This Pull Request has been created because there is currently no description of the `ActiveRecord::DelegatedType` feature in the rails guide.
I thought this feature would be useful for rails users and wanted to add it to the guide.

### Detail

This Pull Request changes Added guide for `ActiveRecord::DelegatedType`.
The following information was used as a reference:

API document: https://api.rubyonrails.org/classes/ActiveRecord/DelegatedType.html
added features PR: https://github.com/rails/rails/pull/39341

### Checklist

Before submitting the PR make sure the following are checked:

* [x] This Pull Request is related to one change. Changes that are unrelated should be opened in separate PRs.
* [x] There are no typos in commit messages and comments.
* [x] Commit message has a detailed description of what changed and why. If this PR fixes a related issue include it in the commit message. Ex: `[Fix #issue-number]`
* [x] Feature branch is up-to-date with `main` (if not - rebase it).
* [x] Pull request only contains one commit for bug fixes and small features. If it's a larger feature, multiple commits are permitted but must be descriptive.
* [x] Tests are added if you fix a bug or add a feature.
* [x] CHANGELOG files are updated for the changed libraries if there is a behavior change or additional feature. Minor bug fixes and documentation changes should not be included.
* [x] PR is not in a draft state.
* [x] CI is passing.

### CODE REVIEW

File: guides/source/association_basics.md [hunk 1]
```diff
@@ -2724,3 +2724,102 @@ will run a query like:
 ```sql
 SELECT "vehicles".* FROM "vehicles" WHERE "vehicles"."type" IN ('Car')
 ```
+
+Delegated types
+------------------------------
+
```
Comment (vipulnsward) [inline guides/source/association_basics.md:2730]: This starts abruptly, lets add example of what the problem is first?

File: guides/source/association_basics.md [hunk 2]
```diff
@@ -2724,3 +2724,102 @@ will run a query like:
 ```sql
 SELECT "vehicles".* FROM "vehicles" WHERE "vehicles"."type" IN ('Car')
 ```
+
+Delegated types
+------------------------------
```
Comment (zzak) [inline guides/source/association_basics.md:2729]:
```suggestion
Delegated types
-----------------
```

File: guides/source/association_basics.md (hunk 1 above)
Comment (soartec-lab) [reply guides/source/association_basics.md:2730]: ok. Correct the content 👍

Comment (soartec-lab) [reply guides/source/association_basics.md:2744]: @vipulnsward 

Thank you for your review.
After presenting the `STI` problem, changed the explanation to use an example.
How about an explanation after the change?

File: guides/source/association_basics.md (hunk 2 above)
Comment (soartec-lab) [reply guides/source/association_basics.md]: Fixed

https://github.com/rails/rails/pull/46190/files#diff-d977574b785ce95fe656c58211c668992bf95214d52d3ffadb409ef8f69ad287R2729

**Comment by soartec-lab** [comment]
@vipulnsward @zzak 

Corrected the pointed out items.
could you review again?

File: guides/source/association_basics.md [hunk 3]
```diff
@@ -2724,3 +2724,156 @@ will run a query like:
 ```sql
 SELECT "vehicles".* FROM "vehicles" WHERE "vehicles"."type" IN ('Car')
 ```
+
+Delegated types
```
Comment (p8) [inline guides/source/association_basics.md:2728]: Headings should be mostly [capitalized](https://edgeguides.rubyonrails.org/ruby_on_rails_guides_guidelines.html#headings).
```suggestion
Delegated Types
```

Comment (soartec-lab) [reply guides/source/association_basics.md]: Fixed

https://github.com/rails/rails/pull/46190/files#diff-d977574b785ce95fe656c58211c668992bf95214d52d3ffadb409ef8f69ad287R2728

File: guides/source/association_basics.md [hunk 4]
```diff
@@ -2724,3 +2724,102 @@ will run a query like:
...
+This is created at the same time by specifying an object of subclass `Message` in the `entryable` of the `Entry` class.
+
+```ruby
+Entry.create! entryable: Message.new(subject: "hello!")
+```
+
+### Adding further delegation
+
+If you want to make further queries on `Entry`, you can define `delegates` and use polymorphism to delegate to subclasses.
+This delegates the `title` method from `Entry` to subclasses.
```
[the remark this answers is not in the document]
Comment (soartec-lab) [reply guides/source/association_basics.md]: Fixed

https://github.com/rails/rails/pull/46190/files#diff-d977574b785ce95fe656c58211c668992bf95214d52d3ffadb409ef8f69ad287R2857

File: guides/source/association_basics.md [hunk 5]
```diff
@@ -2724,3 +2724,102 @@ will run a query like:
...
+  include Entryable
+end
+
+# Schema: comments[ id, content ]
+class Comment < ApplicationRecord
+  include Entryable
+end
+```
+
+This completes the definition of `delegated types`. Entry implements the following methods:
```
[the remark this answers is not in the document]
Comment (soartec-lab) [reply guides/source/association_basics.md]: Fixed

https://github.com/rails/rails/pull/46190/files#diff-d977574b785ce95fe656c58211c668992bf95214d52d3ffadb409ef8f69ad287R2830

File: guides/source/association_basics.md [hunk 6]
```diff
@@ -2724,3 +2724,102 @@ will run a query like:
...
+end
+```
+
+The name of the `delegated_type` definition specifies `entryable`. Also, specified subclasses `Message`, `Comment` as the delegate.
+
+The `Entry` class has `entryable_type` and `entryable_id` fields. This is the field with the `_type`, `_id` suffixes added to the name `entryable` in the `delegated_type` definition.
+`entryable_type` stores the subclass name of the delegatee, and `entryable_id` stores the record id of the delegatee subclass.
+
+
+Next, define a module that implements `delegated types`. Declares the `entryable` specified in the `delegated_type` definition to `has_one` association to `entry`.
```
[the remark this answers is not in the document]
Comment (soartec-lab) [reply guides/source/association_basics.md]: Fixed

https://github.com/rails/rails/pull/46190/files#diff-d977574b785ce95fe656c58211c668992bf95214d52d3ffadb409ef8f69ad287R2804

**Comment by soartec-lab** [comment]
@p8 

Thank you for your review.

Corrected the pointed out items.
could you review again?

File: guides/source/association_basics.md [hunk 7]
```diff
@@ -2724,3 +2724,156 @@ will run a query like:
 ```sql
 SELECT "vehicles".* FROM "vehicles" WHERE "vehicles"."type" IN ('Car')
 ```
+
+Delegated Types
+----------------
+
+[`Single Table Inheritance (STI)`](#single-table-inheritance-sti) works best when there is little difference between subclasses and their attributes, but includes all attributes of all subclasses I need to create a single table.
```
Comment (zzak) [inline guides/source/association_basics.md:2731]: We generally don't use "I" in the documentation, instead we prefer "you", I think.

> but includes all attributes of all subclasses you need to create a single table.

File: guides/source/association_basics.md [hunk 8]
```diff
@@ -2724,3 +2724,156 @@ will run a query like:
...
+
+This has the disadvantage that attributes that are only used by a specific subclass are added to the same table, even though they are unnecessary attributes for another subclass, which bloats a single table.
```
Comment (zzak) [inline guides/source/association_basics.md:2733]: This seems confusing to me, maybe too wordy? Let me try.

```suggestion
The disadvantage of this approach is that it results in bloat to that table. Since it will even include attributes specific to a subclass that aren't used by anything else.
```

File: guides/source/association_basics.md [hunk 9]
```diff
@@ -2724,3 +2724,156 @@ will run a query like:
...
+
+In the following example, the `Comment` does not need a `subject`, but the `Message` need a `subject`, so the `Entry` includes the `subject` attribute.
```
Comment (zzak) [inline guides/source/association_basics.md:2735]:
```suggestion
In the following example, there are two Active Record models that inherit from the same "Entry" class which includes the `subject` attribute.
```

File: guides/source/association_basics.md [hunk 10]
```diff
@@ -2724,3 +2724,156 @@ will run a query like:
...
+end
+
+class Comment < Entry
+end
+
+class Message < Entry
+end
+```
+
+`Delegated types` solves this problem.
```
Comment (zzak) [inline guides/source/association_basics.md:2749]: We generally only backtick stuff that is actual code.

```suggestion
Delegated types solves this problem, via `delegated_type`.
```

File: guides/source/association_basics.md [hunk 11]
```diff
@@ -2724,3 +2724,156 @@ will run a query like:
...
+
+A superclass has a table that stores attributes that are shared among all subclasses.
```
Comment (zzak) [inline guides/source/association_basics.md:2751]: I would rewrite these 3 sentences as such:

In order to use delegated types, we have to model our data in a particular way. The requirements are as follows:


* There is a superclass that stores shared attributes among all subclasses in it's table.
* Each subclass must inherit from the super class, and will have a separate table for any additional attributes specific to it.

This eliminates the need to define attributes in a single table that are unintentionally shared among all subclasses.

File: guides/source/association_basics.md [hunk 12]
```diff
@@ -2724,3 +2724,156 @@ will run a query like:
...
+Each subclass should have a separate table for additional attributes specific to its implementation.
+This eliminates the need to define attributes in one table that are only used by a particular subclass.
+
+First, let's generate the base `Entry` model:
```
Comment (zzak) [inline guides/source/association_basics.md:2755]:
```suggestion
In order to apply this to our example about, we need to generate some new models. First, let's generate the base `Entry` model which will act as our superclass:
```

File: guides/source/association_basics.md [hunk 13]
```diff
@@ -2724,3 +2724,156 @@ will run a query like:
...
+
+```bash
+$ bin/rails generate model entry entryable_type:string entryable_id:integer
+```
+
+The generated model will look like this:
```
Comment (zzak) [inline guides/source/association_basics.md:2761]: I think we can leave out this part because it's generated for us, and seems redundant.

```suggestion
```

File: guides/source/association_basics.md [hunk 14]
```diff
@@ -2724,3 +2724,156 @@ will run a query like:
...
+
+```ruby
+# Schema: entries[ id, entryable_type, entryable_id, created_at, updated_at ]
+class Entry < ApplicationRecord
+end
+```
```
Comment (zzak) [inline guides/source/association_basics.md:2767]:
```suggestion
```

File: guides/source/association_basics.md [hunk 15]
```diff
@@ -2724,3 +2724,156 @@ will run a query like:
...
+
+Then, generate the `Message` and `Comment` models for the delegation:
```
Comment (zzak) [inline guides/source/association_basics.md:2769]:
```suggestion
Then, we will generate new `Message` and `Comment` models for delegation:
```

File: guides/source/association_basics.md [hunk 16]
```diff
@@ -2724,3 +2724,156 @@ will run a query like:
...
+
+```bash
+$ bin/rails generate model message subject:string body:string
+$ bin/rails generate model comment content:string
+```
+
+The generated model will look like this:
```
Comment (zzak) [inline guides/source/association_basics.md:2776]:
```suggestion
The generated models will look like this:
```

Then we can include the superclass "Entry" model and schema below here, I think it makes sense to keep them all together in a single code block and looks nice.

File: guides/source/association_basics.md [hunk 17]
```diff
@@ -2724,3 +2724,156 @@ will run a query like:
...
+class Comment < ApplicationRecord
+end
+```
+
+### Declare `delegated_type`
+
+First, declare a `delegated_type` in the superclass `Entry`.
+
+```ruby
+# Schema: entries[ id, entryable_type, entryable_id, created_at, updated_at ]
```
Comment (zzak) [inline guides/source/association_basics.md:2793]:
```suggestion
```

I think we don't need to repeat the schema every time.

File: guides/source/association_basics.md [hunk 18]
```diff
@@ -2724,3 +2724,156 @@ will run a query like:
...
+class Entry < ApplicationRecord
+  delegated_type :entryable, types: %w[ Message Comment ], dependent: :destroy
+end
+```
+
+The name of the `delegated_type` definition specifies `entryable`. Also, specified subclasses `Message`, `Comment` as the delegate.
```
Comment (zzak) [inline guides/source/association_basics.md:2799]:
```suggestion
The `entryable` parameter specifies the field to use for delegation, and include the types `Message` and `Comment` as the delegate classes.
```

File: guides/source/association_basics.md [hunk 19]
```diff
@@ -2724,3 +2724,156 @@ will run a query like:
...
+
+The `Entry` class has `entryable_type` and `entryable_id` fields. This is the field with the `_type`, `_id` suffixes added to the name `entryable` in the `delegated_type` definition.
+`entryable_type` stores the subclass name of the delegatee, and `entryable_id` stores the record id of the delegatee subclass.
+
+Next, define a module that implements `delegated types`. Declare the `entryable` specified in the `delegated_type` definition to `has_one` association to `entry`.
```
Comment (zzak) [inline guides/source/association_basics.md:2804]:
```suggestion
Next, we must define a module to implement those delegated types, by declaring the `as: :entryable` parameter to the `has_one` association.
```

File: guides/source/association_basics.md [hunk 20]
```diff
@@ -2724,3 +2724,156 @@ will run a query like:
...
+  include Entryable
+end
+
+# Schema: comments[ id, content, created_at, updated_at ]
+class Comment < ApplicationRecord
+  include Entryable
+end
+```
+
+This completes the definition of `delegated types`. `Entry` now implements the following methods:
```
Comment (zzak) [inline guides/source/association_basics.md:2830]:
```suggestion
With this definition complete, our `Entry` delegated type now implements the following methods:
```

File: guides/source/association_basics.md [hunk 21]
```diff
@@ -2724,3 +2724,156 @@ will run a query like:
...
+Entry#message_id      # => returns entryable_id, when entryable_type == "Message", otherwise nil
+Entry.comments        # => Entry.where(entryable_type: "Comment")
+Entry#comment?        # => true when entryable_type == "Comment"
+Entry#comment         # => returns the comment record, when entryable_type == "Comment", otherwise nil
+Entry#comment_id      # => returns entryable_id, when entryable_type == "Comment", otherwise nil
+```
+
+### Object creation
+
+When creating an `Entry` model, create a subclass object at the same time.
```
Comment (zzak) [inline guides/source/association_basics.md:2847]:
```suggestion
When creating a new `Entry` object, we can specify the `entryable` subclass at the same time.
```

File: guides/source/association_basics.md [hunk 22]
```diff
@@ -2724,3 +2724,156 @@ will run a query like:
...
+This is created at the same time by specifying an object of subclass `Message` in the `entryable` of the `Entry` class.
```
Comment (zzak) [inline guides/source/association_basics.md:2848]:
```suggestion
```

File: guides/source/association_basics.md [hunk 23]
```diff
@@ -2724,3 +2724,156 @@ will run a query like:
...
+
+```ruby
+Entry.create! entryable: Message.new(subject: "hello!")
+```
+
+### Adding further delegation
+
+If you want to make further queries on `Entry`, you can define `delegates` and use polymorphism to delegate to subclasses.
```
Comment (zzak) [inline guides/source/association_basics.md:2856]:
```suggestion
We can expand our `Entry` delegator and enhance further by defining `delegates` and use polymorphism to the subclasses.
```

File: guides/source/association_basics.md [hunk 24]
```diff
@@ -2724,3 +2724,156 @@ will run a query like:
...
+For example, to delegate the `title` method from `Entry` to its subclasses:
```
Comment (zzak) [inline guides/source/association_basics.md:2857]:
```suggestion
For example, to delegate the `title` method from `Entry` to it's subclasses:
```

I always get this wrong :joy:

File: guides/source/association_basics.md [hunk 25]
```diff
@@ -2724,3 +2724,156 @@ will run a query like:
...
+
+```ruby
+# Schema: entries[ id, entryable_type, entryable_id, created_at, updated_at ]
```
Comment (zzak) [inline guides/source/association_basics.md:2860]:
```suggestion
```

I think we only need to show the schema above when defining these classes initially.

File: guides/source/association_basics.md [hunk 26]
```diff
@@ -2724,3 +2724,156 @@ will run a query like:
...
+class Entry < ApplicationRecord
+  delegated_type :entryable, types: %w[ Message Comment ]
+  delegates :title, to: :entryable
+end
+
+# Schema: messages[ id, subject, body, created_at, updated_at ]
```
Comment (zzak) [inline guides/source/association_basics.md:2866]:
```suggestion
```

File: guides/source/association_basics.md [hunk 27]
```diff
@@ -2724,3 +2724,156 @@ will run a query like:
...
+class Message < ApplicationRecord
+  def title
+    subject
+  end
+end
+
+# Schema: comments[ id, content, created_at, updated_at ]
```
Comment (zzak) [inline guides/source/association_basics.md:2873]:
```suggestion
```

File: guides/source/association_basics.md [hunk 28]
```diff
@@ -2724,3 +2724,156 @@ will run a query like:
...
+
+```ruby
+# Schema: entries[ id, entryable_type, entryable_id, created_at, updated_at ]
+class Entry < ApplicationRecord
+  delegated_type :entryable, types: %w[ Message Comment ]
+  delegates :title, to: :entryable
+end
+
+# Schema: messages[ id, subject, body, created_at, updated_at ]
+class Message < ApplicationRecord
```
Comment (zzak) [inline guides/source/association_basics.md:2867]:
```suggestion
class Message < ApplicationRecord
  include Entryable
```

We still need to include this right? It might be confusing if that line is suddenly missing.

File: guides/source/association_basics.md [hunk 29]
```diff
@@ -2724,3 +2724,156 @@ will run a query like:
...
+  def title
+    subject
+  end
+end
+
+# Schema: comments[ id, content, created_at, updated_at ]
+class Comment < ApplicationRecord
```
Comment (zzak) [inline guides/source/association_basics.md:2874]:
```suggestion
class Comment < ApplicationRecord
  include Entryable
```

Here too.

File: guides/source/association_basics.md (hunk 7 above)
Comment (soartec-lab) [reply guides/source/association_basics.md]: Fixed

https://github.com/rails/rails/pull/46190/files#diff-d977574b785ce95fe656c58211c668992bf95214d52d3ffadb409ef8f69ad287R2731

File: guides/source/association_basics.md (hunk 8 above)
Comment (soartec-lab) [reply guides/source/association_basics.md]: thank you. I replaced it with Suggest.

https://github.com/rails/rails/pull/46190/files#diff-d977574b785ce95fe656c58211c668992bf95214d52d3ffadb409ef8f69ad287R2733

File: guides/source/association_basics.md (hunk 9 above)
Comment (soartec-lab) [reply guides/source/association_basics.md]: Fixed

https://github.com/rails/rails/pull/46190/files#diff-d977574b785ce95fe656c58211c668992bf95214d52d3ffadb409ef8f69ad287R2735

File: guides/source/association_basics.md (hunk 10 above)
Comment (soartec-lab) [reply guides/source/association_basics.md]: OK, I removed the backtick in the following part.

https://github.com/rails/rails/pull/46190/files#diff-d977574b785ce95fe656c58211c668992bf95214d52d3ffadb409ef8f69ad287R2749

File: guides/source/association_basics.md (hunk 11 above)
Comment (soartec-lab) [reply guides/source/association_basics.md]: Fixed

https://github.com/rails/rails/pull/46190/files#diff-d977574b785ce95fe656c58211c668992bf95214d52d3ffadb409ef8f69ad287R2751-R2756

File: guides/source/association_basics.md (hunk 12 above)
Comment (soartec-lab) [reply guides/source/association_basics.md]: Fixed

https://github.com/rails/rails/pull/46190/files#diff-d977574b785ce95fe656c58211c668992bf95214d52d3ffadb409ef8f69ad287R2758

File: guides/source/association_basics.md (hunk 13 above)
Comment (soartec-lab) [reply guides/source/association_basics.md]: I deleted this.

File: guides/source/association_basics.md (hunk 14 above)
Comment (soartec-lab) [reply guides/source/association_basics.md]: I deleted this.

File: guides/source/association_basics.md (hunk 15 above)
Comment (soartec-lab) [reply guides/source/association_basics.md]: Fixed

https://github.com/rails/rails/pull/46190/files#diff-d977574b785ce95fe656c58211c668992bf95214d52d3ffadb409ef8f69ad287R2764

File: guides/source/association_basics.md (hunk 16 above)
Comment (soartec-lab) [reply guides/source/association_basics.md]: It sure sounds good.
After enumerating the commands to generate models, we modified the configuration so that all generated models and schema are expressed in the same code block.

https://github.com/rails/rails/pull/46190/files#diff-d977574b785ce95fe656c58211c668992bf95214d52d3ffadb409ef8f69ad287R2759-R2786

File: guides/source/association_basics.md (hunk 17 above)
Comment (soartec-lab) [reply guides/source/association_basics.md]: OK, i have removed these.

File: guides/source/association_basics.md (hunk 18 above)
Comment (soartec-lab) [reply guides/source/association_basics.md]: Fixed 

https://github.com/rails/rails/pull/46190/files#diff-d977574b785ce95fe656c58211c668992bf95214d52d3ffadb409ef8f69ad287R2798

File: guides/source/association_basics.md (hunk 19 above)
Comment (soartec-lab) [reply guides/source/association_basics.md]: Fixed

https://github.com/rails/rails/pull/46190/files#diff-d977574b785ce95fe656c58211c668992bf95214d52d3ffadb409ef8f69ad287R2803

File: guides/source/association_basics.md (hunk 20 above)
Comment (soartec-lab) [reply guides/source/association_basics.md]: Fixed

https://github.com/rails/rails/pull/46190/files#diff-d977574b785ce95fe656c58211c668992bf95214d52d3ffadb409ef8f69ad287R2827

File: guides/source/association_basics.md (hunk 21 above)
Comment (soartec-lab) [reply guides/source/association_basics.md]: Fixed

https://github.com/rails/rails/pull/46190/files#diff-d977574b785ce95fe656c58211c668992bf95214d52d3ffadb409ef8f69ad287R2844

File: guides/source/association_basics.md (hunk 22 above)
Comment (soartec-lab) [reply guides/source/association_basics.md]: i have removed this.

File: guides/source/association_basics.md (hunk 23 above)
Comment (soartec-lab) [reply guides/source/association_basics.md]: Fixed

https://github.com/rails/rails/pull/46190/files#diff-d977574b785ce95fe656c58211c668992bf95214d52d3ffadb409ef8f69ad287R2852

File: guides/source/association_basics.md (hunk 24 above)
Comment (soartec-lab) [reply guides/source/association_basics.md]: Exactly. thanks for noticing.

https://github.com/rails/rails/pull/46190/files#diff-d977574b785ce95fe656c58211c668992bf95214d52d3ffadb409ef8f69ad287R2853

File: guides/source/association_basics.md (hunk 25 above)
Comment (soartec-lab) [reply guides/source/association_basics.md]: The schemas has been modified to display only the first time below and delete all subsequent notations.

https://github.com/rails/rails/pull/46190/files#diff-d977574b785ce95fe656c58211c668992bf95214d52d3ffadb409ef8f69ad287R2774-R2786

File: guides/source/association_basics.md (hunk 26 above)
Comment (soartec-lab) [reply guides/source/association_basics.md]: I have removed this.

File: guides/source/association_basics.md (hunk 27 above)
Comment (soartec-lab) [reply guides/source/association_basics.md]: I have removed this.

Comment (soartec-lab) [reply guides/source/association_basics.md]: I have removed this.

https://github.com/rails/rails/pull/46190/files#diff-d977574b785ce95fe656c58211c668992bf95214d52d3ffadb409ef8f69ad287R2855-R2876

File: guides/source/association_basics.md (hunk 28 above)
Comment (soartec-lab) [reply guides/source/association_basics.md:2861]: indeed. Added `include Entryable` to avoid confusion.

https://github.com/rails/rails/pull/46190/files#diff-d977574b785ce95fe656c58211c668992bf95214d52d3ffadb409ef8f69ad287R2862

File: guides/source/association_basics.md (hunk 29 above)
Comment (soartec-lab) [reply guides/source/association_basics.md:2869]: I added `include` here as well.

https://github.com/rails/rails/pull/46190/files#diff-d977574b785ce95fe656c58211c668992bf95214d52d3ffadb409ef8f69ad287R2870

**Comment by soartec-lab** [comment]
@zzak 

Thank you for your review.
I have corrected what you pointed out and responded to your comment.
could you review again?

File: guides/source/association_basics.md [hunk 30]
```diff
@@ -2724,3 +2724,153 @@ will run a query like:
...
+Delegated types solves this problem, via `delegated_type`.
+
+In order to use delegated types, we have to model our data in a particular way. The requirements are as follows:
+
+* There is a superclass that stores shared attributes among all subclasses in it's table.
+* Each subclass must inherit from the super class, and will have a separate table for any additional attributes specific to it.
+
+This eliminates the need to define attributes in a single table that are unintentionally shared among all subclasses.
+
+In order to apply this to our example about, we need to generate some new models.
```
Comment (zzak) [inline guides/source/association_basics.md:2758]:
```suggestion
In order to apply this to our example above, we need to regenerate our models.
```

File: guides/source/association_basics.md [hunk 31]
```diff
@@ -2724,3 +2724,153 @@ will run a query like:
...
+class Message < ApplicationRecord
+  include Entryable
+end
+
+class Comment < ApplicationRecord
+  include Entryable
+end
+```
+
+With this definition complete, our `Entry` delegated type now implements the following methods:
```
Comment (zzak) [inline guides/source/association_basics.md:2827]:
```suggestion
With this definition complete, our `Entry` delegator now provides the following methods:
```

File: guides/source/association_basics.md (hunk 30 above)
Comment (soartec-lab) [reply guides/source/association_basics.md]: Fixed

https://github.com/rails/rails/pull/46190/files#diff-d977574b785ce95fe656c58211c668992bf95214d52d3ffadb409ef8f69ad287R2758

File: guides/source/association_basics.md (hunk 31 above)
Comment (soartec-lab) [reply guides/source/association_basics.md]: Fixed

https://github.com/rails/rails/pull/46190/files#diff-d977574b785ce95fe656c58211c668992bf95214d52d3ffadb409ef8f69ad287R2827

**Comment by soartec-lab** [comment]
@zzak 

Thanks again for your review.
I have corrected what was pointed out.

**Review by zzak — APPROVED** [review]
Thanks for your patches! :bow:
